apps/api/src/app/widgets/usecases/remove-messages/remove-all-messages.usecase.ts

```diff
@@ -0,0 +1,137 @@
+export class RemoveAllMessages {
+  constructor(
+    private invalidateCache: InvalidateCacheService,
+    private messageRepository: MessageRepository,
+    private wsQueueService: WsQueueService,
+    private analyticsService: AnalyticsService,
+    private subscriberRepository: SubscriberRepository,
+    private memberRepository: MemberRepository,
+    private feedRepository: FeedRepository
+  ) {}
+
+  async execute(command: RemoveAllMessagesCommand): Promise<number> {
+    await this.invalidateCache.invalidateQuery({
+      key: buildFeedKey().invalidate({
+        subscriberId: command.subscriberId,
+        _environmentId: command.environmentId,
+      }),
+    });
+
+    await this.invalidateCache.invalidateQuery({
+      key: buildMessageCountKey().invalidate({
+        subscriberId: command.subscriberId,
+        _environmentId: command.environmentId,
+      }),
+    });
+
+    const subscriber = await this.subscriberRepository.findBySubscriberId(command.environmentId, command.subscriberId);
+    if (!subscriber) throw new NotFoundException(`Subscriber ${command.subscriberId} not found`);
+
+    let deletedMessages;
+    let feed;
+    if (command.feedId !== '' && command.feedId !== null && command.feedId !== undefined) {
+      feed = await this.feedRepository.findById(command.feedId);
+      if (!feed) {
+        throw new NotFoundException(`Feed with ${command.feedId} not found`);
+      }
+    }
+    try {
+      const deleteMessageQuery: Partial<MessageEntity> = {
+        _environmentId: command.environmentId,
+        _organizationId: command.organizationId,
+        _subscriberId: subscriber._id,
+        channel: ChannelTypeEnum.IN_APP,
+      };
+
+      if (feed) {
+        deleteMessageQuery._feedId = feed._id;
+      }
+
+      deletedMessages = await this.messageRepository.deleteMany(deleteMessageQuery);
+
+      await this.updateServices(command, subscriber, MarkEnum.SEEN);
+      await this.updateServices(command, subscriber, MarkEnum.READ);
+
+      const admin = await this.memberRepository.getOrganizationAdminAccount(command.organizationId);
+      if (admin) {
+        this.analyticsService.track(`Removed Message - [Notification Center]`, admin._userId, {
+          _subscriber: subscriber._id,
+          _organization: command.organizationId,
+          _environment: command.environmentId,
+        });
+      }
+    } catch (e) {
+      if (e instanceof DalException) {
+        throw new ApiException(e.message);
+      }
+      throw e;
+    }
+
+    return deletedMessages.modifiedCount;
+  }
+
+  private async updateServices(command: RemoveAllMessagesCommand, subscriber, marked: string) {
+    const admin = await this.memberRepository.getOrganizationAdminAccount(command.organizationId);
+    const count = await this.messageRepository.getCount(
+      command.environmentId,
+      subscriber._id,
+      ChannelTypeEnum.IN_APP,
+      {
+        [marked]: false,
+      },
+      { limit: 1000 }
+    );
```

Comment:
the count will be always here zero if there was no feedId provided, so in that case, we don't need to do this operation...

Reply:
shall I write if(feedId) condition where I am calling this function?

Reply:
> the count will be always here zero if there was no feedId provided, so in that case, we don't need to do this operation...

why? if the user won't provide the feedId won't the query delete all the available messages?

Reply:
> shall I write if(feedId) condition where I am calling this function?

I think that you can pass the feed here and check if it exists and if yes then get the count. And by default set the count to zero, because we should remove all the messages for that case, meaning count will be always zero.